library/src/main/kotlin/com/thefuntasty/donut/DonutProgressView.kt

```diff
@@ -296,8 +345,8 @@ class DonutProgressView @JvmOverloads constructor(
         animationEnded: (() -> Unit)? = null
     ): ValueAnimator {
         return ValueAnimator.ofFloat(line.length, to).apply {
-            duration = ANIM_DURATION_MS
-            interpolator = DecelerateInterpolator(INTERPOLATOR_FACTOR)
+            duration = if (animationEnabled) animationDurationMs else 0L
+            interpolator = animationInterpolator
             addUpdateListener {
                 line.length = it.animatedValue as Float
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/thefuntasty/donut/blob/828b548c8186e32727ef08c33dcf88e758e87083/library/src/main/kotlin/com/thefuntasty/donut/DonutProgressView.kt#L351"></span>it.animatedValue cannot be safely cast to Float.</td>
    </tr>
  </tbody>
</table>